src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Options/OptionsStorage.cs

```diff
@@ -69,7 +69,7 @@ public LogLevel LogLevel
     }
 
     [ImportingConstructor]
-    public OptionsStorage(SVsServiceProvider vsServiceProvider, ITelemetryReporter telemetryReporter)
+    public OptionsStorage(SVsServiceProvider vsServiceProvider, Lazy<ITelemetryReporter> telemetryReporter)
```

Comment:
>OptionsStorage [](http://example.com/codeflow?start=11&length=14)

Not asking for a change here, just a general question. We talked about using primary constructor when we touch the class (especially constructor I guess?) and moving constructor code to field initializers. Should we have done it here? I don't necessarily prefer that and kind of like having the initialization code in one familiar place where it's easy to find and debug through, mostly trying to close on the conversation we had in my last PR as to - do we do primary constructors always, no matter what, or only when they make the code cleaner. Not sure brining in primary constructor here would make the code cleaner, possibly harder to read and understand (possibly not :) ).

Reply:
My approach has been to convert to primary constructors, and explicitly capture the parameters to read-only fields, where the constructor just does simple field initialisation and null checks. This constructor does more than that and would be a bad candidate, and I agree with you, it would be harder to understand to change this one.


Having said that, I havent been super aggressive with it, and if I don't modify a constructor, or get annoyed by how many lines all of the null checks take up, I might not notice it could be updated

Reply:
That sounds reasonable to me.